### ISSUE TITLE
A call to `setTime()` fails with a `struct-error`

### ISSUE THREAD

**Issue by gjr80** [gjr80/weewx-aurora#10]
During normal operation a call to `AuroraDriver.setTime()` fails with a `struct-error` causing WeeWX to exit. Stack trace:
```
Jan 01 12:04:56 aurora weewxd[478209]: INFO __main__: Starting up weewx version 5.0.0b17
Jan 01 12:04:56 aurora weewxd[478209]: INFO weewx.engine: Clock error is 5.11 seconds (positive is fast)
Jan 01 12:04:56 aurora weewxd[478209]: INFO weewx.engine: Main loop exiting. Shutting engine down.
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__: Caught unrecoverable exception:
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****  required argument is not an integer
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****  Traceback (most recent call last):
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****    File "/home/weewxuser/weewx/src/weewxd.py", line 158, in main
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****      engine.run()
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****    File "/home/weewxuser/weewx/src/weewx/engine.py", line 174, in run
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****      self.dispatchEvent(weewx.Event(weewx.STARTUP))
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****    File "/home/weewxuser/weewx/src/weewx/engine.py", line 241, in dispatchEvent
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****      callback(event)
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****    File "/home/weewxuser/weewx/src/weewx/engine.py", line 770, in startup
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****      self.do_sync()
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****    File "/home/weewxuser/weewx/src/weewx/engine.py", line 793, in do_sync
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****      self.engine.console.setTime()
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****    File "/home/weewxuser/weewx-data/bin/user/aurora.py", line 647, in setTime
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****      _response = self.inverter.set_time(_inv_ts_b)
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****    File "/home/weewxuser/weewx-data/bin/user/aurora.py", line 1762, in set_time
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****      response_t = self.execute_cmd_with_crc('set_time_date', p1=inverter_ts)
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****    File "/home/weewxuser/weewx-data/bin/user/aurora.py", line 1132, in execute_cmd_with_crc
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****      _command_bytes_crc = self.construct_cmd_message(command_code=self.commands[command]['cmd_code'],
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****    File "/home/weewxuser/weewx-data/bin/user/aurora.py", line 1424, in construct_cmd_message
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****      command_bytes = struct.pack('8B', *padded_command_t)
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****  struct.error: required argument is not an integer
Jan 01 12:04:56 aurora weewxd[478209]: CRITICAL __main__:     ****  Exiting.
```

**Comment by gjr80** [comment]
`setTime()` has worked for considerable time in releases before v0.7.0. Issue is likely due to re-factoring of inverter command construct code during v0.7.0 development.

**Comment by gjr80** [comment]
fixed in v0.7.0b2
